llvm/lib/Target/AMDGPU/AMDGPUMCResourceInfo.cpp

```diff
@@ -164,6 +167,8 @@ void MCResourceInfo::gatherResourceInfo(
     for (const Function *Callee : FRI.Callees) {
       if (!Seen.insert(Callee).second)
         continue;
+      if (!MF.getFunction().doesNotRecurse() && !Callee->doesNotRecurse())
```

Comment:
Is this structurally relying on the norecurse attribute? You can't rely on it too much, you still need to not blow up in the undefined behavior case.

Unfortunately this will also miss when norecurse is present on the callsite, but not the declaration